File: kotlin/app/src/main/java/com/example/cityexplorer/data/dtos/UserDtos.kt
package com.example.cityexplorer.data.dtos

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ValidateLoginTokenRequestDto(
    @SerialName("googleToken")
    val googleToken: String
)

@Serializable
data class ValidateLoginTokenResponseDto(
    @SerialName("isSuccess")
    val isSuccess: Boolean,

    @SerialName("token")
    val token: String
)

@Serializable
data class ValidateAuthorizationTokenResponseDto(
    @SerialName("id")
    val id: String = ""
)

@Serializable
data class GetUserStatisticsResponseDto(
    @SerialName("explored")
    val explored: Double,

    @SerialName("progress")
    val progress: Int,

    @SerialName("hexagonCount")
    val hexagonCount: Int,

    @SerialName("playTime")
    val playTime: Int,

    @SerialName("distance")
    val distance: Int,

    @SerialName("ranking")
    val ranking: Int,

    @SerialName("userCount")
    val userCount: Int
)

File: csharp/api/Dtos/UserDtos.cs
using System.Text.Json.Serialization;

namespace csharp.Dtos;

public class ValidateLoginTokenRequestDto
{
    [JsonPropertyName("googleToken")] public required string GoogleToken { get; init; }
}

public class ValidateLoginTokenResponseDto
{
    [JsonPropertyName("isSuccess")] public bool IsSuccess { get; init; }

    [JsonPropertyName("token")] public required string Token { get; init; }
}

public class ValidateAuthorizationTokenResponseDto
{
    [JsonPropertyName("id")] public string Id { get; init; } = string.Empty;
}

public class GetUserStatisticsResponseDto
{
    [JsonPropertyName("explored")] public double Explored { get; init; }

    [JsonPropertyName("progress")] public int Progress { get; init; }

    [JsonPropertyName("hexagonCount")] public int HexagonCount { get; init; }

    [JsonPropertyName("playTime")] public int PlayTime { get; init; }

    [JsonPropertyName("distance")] public int Distance { get; init; }

    [JsonPropertyName("ranking")] public int Ranking { get; init; }

    [JsonPropertyName("userCount")] public int UserCount { get; init; }
}